File: mobile/src/utils/validators.ts
/**
 * 共享校验工具:手机号、验证码等正则。
 *
 * 校验时记得先 .trim() 掉首尾空格 — 输入框常会自动加空格。
 */

export const phoneRegex = /^1[3-9]\d{9}$/;
export const codeRegex = /^\d{6}$/;

export const isPhone = (s: string) => phoneRegex.test(s.trim());
export const isCode = (s: string) => codeRegex.test(s.trim());

File: android/app/src/main/java/com/jueqiao/jianghu/data/Validators.kt
package com.jueqiao.jianghu.data

/**
 * Validation utilities — mirrors utils/validators.ts.
 */
object Validators {
    private val phoneRegex = Regex("^1[3-9]\\d{9}$")
    private val codeRegex  = Regex("^\\d{6}$")

    fun isPhone(raw: String): Boolean = phoneRegex.matches(raw.trim())
    fun isCode(raw: String): Boolean  = codeRegex.matches(raw.trim())

    /** Passwords are passed through exactly as entered; never trim secret values. */
    fun isPassword(raw: String): Boolean = raw.length in 8..64
}
